sale_milestone_profile_invoicing/models/sale_order_line.py

```diff
@@ -0,0 +1,129 @@
+                rate_line = ts.project_id.sale_line_employee_ids.filtered(
+                    lambda r: r.employee_id == ts.employee_id
+                )
+                total += ts.unit_amount * rate_line.price_unit
+            line.amount_delivered_from_task = total
+            line.amount_delivered_from_task_company_currency = (
+                total * line.order_id.currency_rate
+            )
+
+    @api.multi
+    @api.depends('task_id', 'task_id.timesheet_ids.timesheet_invoice_id')
+    def _compute_amount_invoiced_from_task(self):
+        for line in self:
+            total = 0
+            for ts in line._get_timesheet_for_amount_calculation(True):
+                rate_line = ts.project_id.sale_line_employee_ids.filtered(
+                    lambda r: r.employee_id == ts.employee_id
+                )
+                total += ts.unit_amount * rate_line.price_unit
+            line.amount_invoiced_from_task = total
+            line.amount_invoiced_from_task_company_currency = (
+                total * line.order_id.currency_rate
+            )
+
+    @api.multi
+    @api.depends('amount_delivered_from_task', 'product_uom_qty', 'price_unit')
+    def _compute_qty_delivered(self):
+        """Change qantity delivered for line with a product milestone."""
+        super()._compute_qty_delivered()
+        for line in self:
+            if line._is_linked_to_milestone_product():
+                line.qty_delivered = (
+                    line.product_uom_qty
+                    * line.amount_delivered_from_task
+                    / line.price_unit
+                )
+
+    @api.multi
+    @api.depends('amount_invoiced_from_task', 'product_uom_qty', 'price_unit')
+    def _get_invoice_qty(self):
+        """Change qantity invoiced for line with a product milestone."""
+        super()._get_invoice_qty()
+        for line in self:
+            if line._is_linked_to_milestone_product():
+                line.qty_invoiced = (
+                    line.product_uom_qty
+                    * line.amount_invoiced_from_task
+                    / line.price_unit
+                )
+
+    @api.multi
+    def _is_linked_to_milestone_product(self):
+        self.ensure_one()
+        return (
+            self.product_id.type == 'service'
+            and self.product_id.service_policy == 'delivered_manual'
+            and self.product_id.service_tracking == 'task_new_project'
+        )
+
+    @api.multi
+    def _compute_invoice_status(self):
+        """Change invoice status for milestone line"""
+        super()._compute_invoice_status()
+        for line in self:
+            if line._is_linked_to_milestone_product():
+                mapping = line.project_id.sale_line_employee_ids.filtered(
+                    lambda r: r.sale_line_id == line.id
+                )
+                if not mapping:
+                    if line.qty_delivered <= line.product_uom_qty:
```

Comment:
@TDu  No need to `Else` part
```
line.invoice_status = 'upselling'
if line.qty_delivered <= line.product_uom_qty:
    line.invoice_status = 'no'
```

Reply:
I tried but the test cases fail if I do so